src/pickup.cpp

```diff
@@ -41,6 +41,18 @@ bool Pickup::collidesWith(Entity &ent, VECTOR2 &collisionVector) {
 			msgPtr = new Message(messageNS::EFFECT, messageNS::BALL, effectType, duration);
 			setMessage(msgPtr);
 		}	break;
+
+		case entityNS::PADDLE: {
+			setActive(false);
+			if (ent.getX() >= GAME_WIDTH / 2) {
+				msgPtr = new Message(messageNS::EFFECT, messageNS::RIGHT_P, effectType, duration);
+				setMessage(msgPtr);
+			}
+			else {
+				msgPtr = new Message(messageNS::EFFECT, messageNS::LEFT_P, effectType, duration);
+				setMessage(msgPtr);
+			}
```

Comment:
Can you combine them? The only thing that's different is the target type

```c++
messageNS::TARGET_TYPE targetType = (ent.get() > GAME_WIDTH / 2) ? messageNS::RIGHT_P : messageNS::LEFT_P;
```

This line should suffice. Then you init your message with `targetType`